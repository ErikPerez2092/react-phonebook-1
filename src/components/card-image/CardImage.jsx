import React from 'react';
import PropTypes from 'prop-types';

const CardImage = ({ image, title }) => (
  <div>
    <img alt={title} src={image} />
    <h2>{title}</h2>
  </div>
);

CardImage.propTypes = {
  image: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
};

export default CardImage;
