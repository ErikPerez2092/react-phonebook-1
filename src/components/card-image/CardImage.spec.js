import React from 'react';
import { shallow } from 'enzyme';
import CardImage from './CardImage';
import FriendsImage from '../login/friends.jpeg';

describe('CardImageComponent', () => {
  let cardImageComponent;
  const title = 'react-phonebook!';
  beforeEach(() => {
    cardImageComponent = shallow(<CardImage title={title} image={FriendsImage} />);
  });

  it('sould render <CardImage />', () => {
    expect(cardImageComponent.getElements()).toMatchSnapshot();
  });

  fit('should contain an image', () => {
    expect(cardImageComponent.find('img')).toHaveLength(1);
    const { alt, src } = cardImageComponent.find('img').props();
    expect(alt).toBe(title);
    expect(src).toBe(FriendsImage);
  });

  it('should have a title', () => {
    expect(cardImageComponent.find('h2')).toHaveLength(1);
    expect(cardImageComponent.find('h2').text()).toBe(title);
  });
});
