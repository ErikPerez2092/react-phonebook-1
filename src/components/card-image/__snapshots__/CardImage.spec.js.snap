// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardImageComponent sould render <CardImage /> 1`] = `
Array [
  <div>
    <img
        alt="react-phonebook!"
        src="friends.jpeg"
    />
    <h2>
        react-phonebook!
    </h2>
</div>,
]
`;
